libs/utils/utils.js

```diff
@@ -357,6 +432,27 @@ function getExtension(path) {
   return pageName.includes('.') ? pageName.split('.').pop() : '';
 }
 
+function getPrefixBySite(locale, url, relative) {
```

Comment:
I personally find the naming in this function still un-intuitive... if you or another reviewer has some ideas, that's one area where I still see potential, that'll make our lives easier in the future to understand what's going on.

As you mentioned, it's a rather small but very impactful piece of our core logic so everything that happens here should be crystal clear and working with locales, languages, maps becomes very confusing.

Reply:
Naming is really hard to get right, I agree. I also don't think it's perfect, but also don't have better ideas so am open to suggestions.